pgjdbc/src/main/java/org/postgresql/jdbc/BooleanTypeUtil.java

```diff
@@ -43,7 +68,7 @@ static boolean castToBoolean(final Object in) throws PSQLException {
       return fromString((String) in);
     }
     if (in instanceof Character) {
-      return fromCharacter((Character) in);
+      return fromString(in.toString());
```

Comment:
It might be better to have a separate map. I'm not sure how often Character is used to set bools, however `in.toString()` would likely to ruin the performance here.